Sources/Networking/SessionDelegate.swift

```diff
@@ -259,4 +261,43 @@ extension SessionDelegate: URLSessionDataDelegate {
         remove(sessionTask)
         sessionTask.onTaskDone.call((result, sessionTask.callbacks))
     }
+    
+    
+    // MARK: - extraHandler
+    @available(iOS 7.0, OSX 11.0, tvOS 9.0, watchOS 2.0, *)
+    func urlSessionDidFinishEvents(forBackgroundURLSession session: URLSession) {
+        extraHandler?.urlSessionDidFinishEvents?(forBackgroundURLSession: session)
+    }
+    func urlSession(_ session: URLSession, didBecomeInvalidWithError error: Error?) {
+        extraHandler?.urlSession?(session, didBecomeInvalidWithError: error)
+    }
+    @available(iOS 11.0, OSX 10.13, tvOS 11.0, watchOS 4.0, *)
+    func urlSession(_ session: URLSession, taskIsWaitingForConnectivity task: URLSessionTask) {
+        extraHandler?.urlSession?(session, taskIsWaitingForConnectivity: task)
+    }
+    @available(iOS 10.0, OSX 10.12, tvOS 10.0, watchOS 3.0, *)
+    func urlSession(_ session: URLSession, task: URLSessionTask, didFinishCollecting metrics: URLSessionTaskMetrics) {
+        extraHandler?.urlSession?(session, task: task, didFinishCollecting: metrics)
+    }
+    @available(iOS 9.0, OSX 10.11, tvOS 9.0, watchOS 2.0, *)
+    func urlSession(_ session: URLSession, dataTask: URLSessionDataTask, didBecome streamTask: URLSessionStreamTask) {
+        extraHandler?.urlSession?(session, dataTask: dataTask, didBecome: streamTask)
+    }
+    func urlSession(_ session: URLSession, dataTask: URLSessionDataTask, didBecome downloadTask: URLSessionDownloadTask) {
+        extraHandler?.urlSession?(session, dataTask: dataTask, didBecome: downloadTask)
+    }
+    func urlSession(_ session: URLSession, task: URLSessionTask, needNewBodyStream completionHandler: @escaping (InputStream?) -> Void) {
+        extraHandler?.urlSession?(session, task: task, needNewBodyStream: completionHandler)
```

Comment:
If you implements the URLSessionDelegate method with completionBlock, you `MUST` callback the handler, even if you don't have anything logic.